<a href="https://colab.research.google.com/github/ubsuny/PHY386/blob/Examples/Calculate_the_volume_of_a_sphere_with_python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Calculate the volume of a sphere with python

### Cell 1: Problem Explanation
### Volume of a Sphere
We want to compute the volume of a sphere with radius $R$.

#### Formula for the Volume of a Sphere
The formula for the volume of a sphere is given by:
$$
V = \frac{4}{3} \pi R^3
$$
where:
- $R$ is the radius of the sphere,

#### Objective
- Derive the formula using calculus (integration).
- Calculate the volume numerically using Python.


### Derivation of the Formula using Calculus
We can derive the formula for the volume of a sphere using the concept of integration.

The volume of a sphere is computed by summing up the infinitesimally thin circular disks along the $z$-axis from $-R$ to $R$.

##### Step 1: Equation of a Circle
The equation of a circle in the $xz$-plane is:
$$
x^2 + z^2 = R^2
$$
Solving for $x$, the radius of a circular slice at a given $z$, we get:
$$
x = \sqrt{R^2 - z^2}
$$

##### Step 2: Area of the Disk
The area of each disk is:
$$
A(z) = \pi x^2 = \pi (R^2 - z^2)
$$

### Step 3: Integrate to Find the Volume
The volume of the sphere is:
$$
V = \int_{-R}^{R} A(z) \, dz = \int_{-R}^{R} \pi (R^2 - z^2) \, dz
$$

We will compute this integral in Python.


In [1]:
# Importing Required Libraries
import sympy as sp

# Define symbols
R, z = sp.symbols("R z", positive=True)

# Define the integrand for the volume
integrand = sp.pi * (R**2 - z**2)
integrand

pi*(R**2 - z**2)

In [2]:
# Solving the Integral
# Compute the definite integral from -R to R
volume_integral = sp.integrate(integrand, (z, -R, R))
volume_integral

4*pi*R**3/3

In [3]:
# Simplify the Result
# Simplify the result to obtain the standard formula for the volume of a sphere
volume_formula = sp.simplify(volume_integral)
volume_formula

4*pi*R**3/3

### Numerical Calculation
Let's calculate the volume of a sphere for a specific radius $R$. For example, $R = 5$.


In [4]:
# Assign a value to the radius
radius_value = 5

# Evaluate the volume using the formula
numerical_volume = volume_formula.subs(R, radius_value)
f"Volume of a sphere with radius {radius_value} is approximately {numerical_volume:.2f} cubic units."

'Volume of a sphere with radius 5 is approximately 523.60 cubic units.'

### Summary
1. The formula for the volume of a sphere was derived using calculus:
$$
V = \frac{4}{3} \pi R^3
$$
2. Using $R = 5$, the calculated volume is approximately 523.60 cubic units.


### using spherical coordinates

#### Volume of a Sphere Using Spherical Coordinates
We want to compute the volume of a sphere with radius $R$ using spherical coordinates.

#### Spherical Coordinates
In spherical coordinates, the volume element $dV$ is given by:
$$
dV = r^2 \sin\theta \, dr \, d\theta \, d\phi
$$
where:
- $r$ is the radial distance,
- $\theta$ is the polar angle (measured from the positive $z$-axis),
- $\phi$ is the azimuthal angle (measured in the $xy$-plane).

##### Volume of the Sphere
To find the volume of a sphere of radius $R$, we integrate $dV$ over the appropriate limits:
$$
V = \int_0^{2\pi} \int_0^{\pi} \int_0^R r^2 \sin\theta \, dr \, d\theta \, d\phi
$$

#### Objective
- Derive the volume formula for a sphere using spherical coordinates.
- Compute the volume numerically for a specific radius $R$.


In [5]:
# Importing Required Libraries
import sympy as sp

# Define symbols
r, theta, phi, R = sp.symbols("r theta phi R", positive=True)

# Define the volume element in spherical coordinates
dV = r**2 * sp.sin(theta)  # Volume element r^2 sin(theta) dr dtheta dphi
dV

r**2*sin(theta)

#### Setting Up the Triple Integral
The total volume of the sphere is:
$$
V = \int_0^{2\pi} \int_0^{\pi} \int_0^R r^2 \sin\theta \, dr \, d\theta \, d\phi
$$
We will compute this integral step by step.


In [6]:
# Step 1: Integrate over r (radial coordinate)
integrand_r = sp.integrate(r**2, (r, 0, R))
integrand_r

R**3/3

### Integrate Over θ

Next, integrate the result over $\theta$:
$$
\int_0^{\pi} \sin\theta \, d\theta
$$

In [7]:
# Step 2: Integrate over θ (polar angle)
integrand_theta = sp.integrate(sp.sin(theta), (theta, 0, sp.pi))
integrand_theta

2

### Integrate Over φ

Finally, integrate over $\phi$:
$$
\int_0^{2\pi} d\phi
$$

In [8]:
# Step 3: Integrate over φ (azimuthal angle)
integrand_phi = sp.integrate(1, (phi, 0, 2 * sp.pi))
integrand_phi

2*pi

#### Combining Results
Multiply the results of the three integrals to compute the total volume of the sphere.


In [9]:
# Combine all results
volume_sphere = integrand_r * integrand_theta * integrand_phi
volume_sphere.simplify()

4*pi*R**3/3

#### Numerical Calculation
Let's calculate the volume of a sphere for a specific radius $R$. For example, $R = 5$.

In [10]:
# Assign a value to the radius
radius_value = 5

# Substitute the radius into the formula
numerical_volume = volume_sphere.subs(R, radius_value)
f"Volume of a sphere with radius {radius_value} is approximately {numerical_volume:.2f} cubic units."

'Volume of a sphere with radius 5 is approximately 523.60 cubic units.'

### Summary
1. The formula for the volume of a sphere derived using spherical coordinates is:
$$
V = \frac{4}{3} \pi R^3
$$
2. Using $R = 5$, the calculated volume is approximately 523.60 cubic units.